core/Updates/4.0.0-b1.php

```diff
@@ -109,6 +111,18 @@ public function getMigrations(Updater $updater)
         // remove old options
         $migrations[] = $this->migration->db->sql('DELETE FROM `' . Common::prefixTable('option') . '` WHERE option_name IN ("geoip.updater_period", "geoip.loc_db_url", "geoip.isp_db_url", "geoip.org_db_url")');
 
+        // replace days to ... dimensions w/ segments
+        foreach (['log_visit', 'log_conversion'] as $table) {
```

Comment:
btw we should do this all in one query and not in multiple queries so we group all log_visit and all log_conversion update queries.

Reply:
one UPDATE? sure, makes sense

Reply:
@diosmosis are these ones actually needed? I would assume Matomo installs the columns automatically (which is also why you didn't need to add it for new installs to the mysql schema)? 
![image](https://user-images.githubusercontent.com/273120/81629269-d8958f80-9456-11ea-889d-4952918d39f9.png)

Reply:
oh wait I see it is needed for the update... statement, makes sense... Could technically also record that the component was updated so Matomo won't try to update the columns

![image](https://user-images.githubusercontent.com/273120/81629385-1e525800-9457-11ea-8a40-85d5c5ae6a39.png)

like `insert into option values option_value = $columnType, option_name = version_log_visit_ visitor_seconds_since_first `

Reply:
Is this the desired approach?

Reply:
actually I think the dimensions end up getting added after the main update, so we wouldn't be able to initialize the new dimension values

Reply:
yes that's what I figured too. It's needed to add them here already. Be just good to additionally record the component version so once the DB update completed it won't try to change the type of these columns but it might not be too important since it would get the same type anyway and therefore mysql wouldn't do anything.